tokens/models/tokens/tokens_erc20.sql

```diff
@@ -1,34 +1,80 @@
-UNION ALL
-SELECT 'base' as blockchain, contract_address, symbol, decimals FROM {{ ref('tokens_base_erc20') }}
-UNION ALL
-SELECT 'celo' as blockchain, contract_address, symbol, decimals FROM {{ ref('tokens_celo_erc20') }}
-UNION ALL
-SELECT 'zksync' as blockchain, contract_address, symbol, decimals FROM {{ ref('tokens_zksync_erc20') }}
-UNION ALL
-SELECT 'zora' as blockchain, contract_address, symbol, decimals FROM {{ ref('tokens_zora_erc20') }}
-UNION ALL
-SELECT 'scroll' as blockchain, contract_address, symbol, decimals FROM {{ ref('tokens_scroll_erc20') }}
+/*
+    the main source for erc20 tokens is dune.definedfi.dataset_tokens -- an automated source pulled into Dune
+    in order to maintain the same amount of coverage of tokens as before, and provide future addition framework, each chain still contains a static file to add for any missing in the automated source
+*/
+
+{% set static_models = {
+    'tokens_arbitrum': {'blockchain': 'arbitrum', 'model': ref('tokens_arbitrum_erc20')}
+    ,'tokens_avalanche_c': {'blockchain': 'avalanche_c', 'model': ref('tokens_avalanche_c_erc20')}
+    ,'tokens_base': {'blockchain': 'base', 'model': ref('tokens_base_erc20')}
+    ,'tokens_bnb': {'blockchain': 'bnb', 'model': ref('tokens_bnb_bep20')}
+    ,'tokens_celo': {'blockchain': 'celo', 'model': ref('tokens_celo_erc20')}
+    ,'tokens_ethereum': {'blockchain': 'ethereum', 'model': ref('tokens_ethereum_erc20')}
+    ,'tokens_fantom': {'blockchain': 'fantom', 'model': ref('tokens_fantom_erc20')}
+    ,'tokens_gnosis': {'blockchain': 'gnosis', 'model': ref('tokens_gnosis_erc20')}
+    ,'tokens_goerli': {'blockchain': 'goerli', 'model': ref('tokens_goerli_erc20')}
+    ,'tokens_optimism': {'blockchain': 'optimism', 'model': ref('tokens_optimism_erc20')}
+    ,'tokens_polygon': {'blockchain': 'polygon', 'model': ref('tokens_polygon_erc20')}
+    ,'tokens_scroll': {'blockchain': 'scroll', 'model': ref('tokens_scroll_erc20')}
+    ,'tokens_zksync': {'blockchain': 'zksync', 'model': ref('tokens_zksync_erc20')}
+    ,'tokens_zora': {'blockchain': 'zora', 'model': ref('tokens_zora_erc20')}
+} %}
+
+with
+  automated_source as (
+    /*
+        todo: find out how to use source jinja for uploaded data with three namespaces
+        todo: move evms schema to a subproject, establish as a source, otherwise we can't call as ref() here in this subproject
+    */
+    select
+      i.blockchain
+      , t.address as contract_address
+      , t.symbol
+      , t.decimals
+    from
+      dune.definedfi.dataset_tokens as t
+      join evms.info as i on t.networkid = i.chain_id
+), static_source as (
+    select *
+    from
+    (
+        {% for key, value in static_models.items() %}
+        select
+            '{{ value.blockchain }}' as blockchain
+            , contract_address
+            , symbol
+            , decimals
+        from
+            {{ value.model }}
+        {% if value.blockchain == 'optimism' %}
```

Comment:
Could you explain why this is needed?

Reply:
optimism lineage for erc20 tokens is a combo of manually curated and dynamic queries. it differs from all other chains. i don't want to mess with that logic until the new design is in place, then we can come back to optimism to simplify as much as possible
cc @MSilb7